test/integration/job/job_test.go

```diff
@@ -4475,11 +4523,7 @@ func removePodsFinalizer(ctx context.Context, t *testing.T, clientSet clientset.
 		t.Fatalf("Failed to list pods: %v", err)
 	}
 	updatePod(ctx, t, clientSet, pods.Items, func(pod *v1.Pod) {
-		for i, finalizer := range pod.Finalizers {
```

Comment:
What does this change do?

Reply:
Sorry @dejanzele , I didn't ping you yet because this is still WIP. I got problems with a couple of tests which I discussed with @mimowo on Slack. I'm currently working in the fix (which will outdate this change). I'll assign you as soon as I push the changes.